doc/index.md

```diff
@@ -0,0 +1,92 @@
+For even destination registers other than `X0`,  `Xn` and `Xn+1` are reserved for write-back upon offloading a dual write-back instruction, where `Xn` denotes the destination register addresss extracted from `instr_data[11:7]`.
+
+### Hierarchical Interconnect
+The accelerator interface is designed to enable a number of flexible topologies.
+The simplest topology is the direct attachment of one or multiple accelerator to a single CPU core.
+The interconnect also supports sharing of accelerators accross multiple accelerators accross multiple cores in a cluster-like topology.
+The sharing granularity of accelerators is flexible.
+Any number of cores in a cluster can be connected to share a selection of accelerators resulting in a hierarchical interconnect.
+
+
+### Multiple Accelerators of the Same Type (TODO)
+The depicted interconnect schematic relies upon the [stream\_xbar](https://github.com/pulp-platform/common_cells/blob/master/src/stream_xbar.sv) IP to facilitate routing of requests and responses from and to the accelerator structures.
+One limitation of using this IP is that it is not possible to utilize multiple accelerators of the same type at the same accelerator address.
+This issue will be relieved by implementing a variant of the IP which uses a N:K streaming arbiter in the output path of the stream\_xbar.
+An according IP exists in the [APU Cluster](https://github.com/pulp-platform/apu_cluster/tree/master/sourcecode/marx).
+
+## Architecture
+
+### Accelerator Adapter Module
+The accelerator adapter module facilitates accelerator-agnostic instruction offloading to the accelerator interconnect.
+The core-side connection implements the instruction offloading [X-interface](x-interface.md).
+The accelerator adapter connects to the interconnect through the [C-interface](c-interface.md).
+In addition, the adapter module connects to a collection of accelerator-specific predecoder modules.
+
+![Accelerator Adapter](img/acc-adapter.svg)
+
+### Accelerator Interconnect module
+The [accelerator interconnect module](../rtl/acc_interconnect) implements the interconnect fabric on each level of the interconnect hierarchy.
+It comprises a crossbar for routing of requests and responses from a number of requesting units the accelerator structures residing on the corresponding interconnect level, as well as a bypass-path to forward requests from and to a higher hierarchy level.
+All in- and output ports implement the [C-interface](c-interface.md).
+For request and response path, separate pipeline registers may be implemented for each interconnect module.
+
+![Accelerator Interconnect Level](img/acc-interconnect-level.svg)
+
+
+## Requirements
+
+### CPU Support
+In order to continue normal operation during the execution of offloaded instructions, the core must maintain a scoreboard to keep track of outstanding accelerator write-backs.
```

Comment:
I would argue that this isn't really part of the specification, right? It is up to the core implementor to decide how she wants to handle this, right? We could leave it as a comment?

Reply:
Yes, you are right. We could just strictly leave it at the interface definition.